tokio-tls/src/lib.rs

```diff
@@ -20,10 +20,21 @@
 //! built. Configuration of TLS parameters is still primarily done through the
 //! `native-tls` crate.
 
-use futures::{Async, Future, Poll};
 use native_tls::{Error, HandshakeError};
+use std::future::Future;
 use std::io::{self, Read, Write};
-use tokio_io::{try_nb, AsyncRead, AsyncWrite};
+use std::marker::Unpin;
+use std::pin::Pin;
+use std::task::{Context, Poll, Waker};
+use tokio_io::{AsyncRead, AsyncWrite};
+
+scoped_tls::scoped_thread_local!(static WAKER: Waker);
```

Comment:
I realized when updating tokio-openssl that the thread-local approach here has troubles with reentrancy (e.g. weird things like `TlsStream<TlsStream<TcpStream>>)`.

Instead, I'm passing the context pointer (as a usize) directly down into the wrapper around the stream: https://github.com/sfackler/tokio-openssl/blob/tokio-02/src/lib.rs#L69

Reply:
thanks! updated :)

Reply:
It doesn't look like this thread-local is used anymore. You can probably remove the `scoped-tls` dependency as well now.